modules/relayers/src/lib.rs

```diff
@@ -102,9 +109,194 @@ pub mod pallet {
 				},
 			)
 		}
+
+		/// Register relayer or update its registration.
+		///
+		/// Registration allows relayer to get priority boost for its message delivery transactions.
+		#[pallet::call_index(1)]
+		#[pallet::weight(Weight::zero())] // TODO: https://github.com/paritytech/parity-bridges-common/issues/2033
+		pub fn register(origin: OriginFor<T>, valid_till: T::BlockNumber) -> DispatchResult {
```

Comment:
Yep, seems like a good idea.

> I'm not familiar with slashing

Our slashing is not some other slashing. I'm not familiar with other slashing types as well.

Reply:
OK, on a second thought it isn't as simple as it seems. Relayer deregisteration = storage updates (registration removal and balances update). So we can't simply call `deregister` and something will happen after  `required_registration_lease()` blocks. So we need to do `unreserve` and registration removal after this period. It can be made either using additional call or from `on_initialize`/`on_idle`. Both options are worse than the current one imo.

Please check my thoughts - may be I misunderstood you, or something is wrong with my thoughts

Reply:
Oh, yes, you're right. Thinking.